opm/simulators/flow/Main.hpp

```diff
@@ -537,6 +536,22 @@ class Main
         }
     }
 
+    // This function is a special case, if the program has been invoked
+    // with the argument "--test-split-communicator=true" as the FIRST
+    // argument, it will be removed from the argument list and we set the
+    // test_split_comm_ flag to true.
+    // Note: initializing the parameter system before MPI could make this
+    // use the parameter system instead.
+    void handleTestSplitCommunicatorCmdLine_()
+    {
+        if (argc_ >= 2 && std::strcmp(argv_[1], "--test-split-communicator=true") == 0) {
+            test_split_comm_ = true;
+            --argc_;             // We have one less argument.
+            argv_[1] = argv_[0]; // What used to be the first proper argument now becomes the command argument.
+            ++argv_;             // Pretend this is what it always was.
```

Comment:
I am a bit surprised that this works, but maybe I just do not get what the code does.

To me it seems that it converts `flow --test-split-communicator=true model.data` to `flow flow`. As this (hopefully) can't work, I must be missing something.

Why not just insist on `--test-split-communicator=true` being the last argument. (ok, still rather hacky...)?

Reply:
With `flow --test-split-communicator=true model.data` as data, before this we have argc_ == 3 and:
| argv_ position | value |
| --- | --- |
| 0 | flow |
| 1 | --test-split-communicator=true model.data |
| 2 | model.data |

After line 549 argc_ == 2. After line 550:
| argv_ position | value |
| --- | --- |
| 0 | flow |
| 1 | flow |
| 2 | model.data |

After line 551:
| argv_ position | value |
| --- | --- |
| 0 | flow |
| 1 | model.data |

Reply:
:flushed: Well that change I did miss.